ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java

```diff
@@ -335,6 +347,13 @@ long getMaxBlobSize() {
       retValue =
           new Pair<>(referenceTimeInMs, collectValidDataSizeByLogSegment(referenceTimeInMs, expiryReferenceTime));
     }
+    if (fillupLogSegmentNames) {
+      // Before return the value, make sure all the log segments are in the final map
+      for (IndexSegment indexSegment : index.getIndexSegments().descendingMap().values()) {
```

Comment:
`index` has access to `log`,  and `log` has a map `segmentsByName`. Maybe we can expose some API in PersistentIndex to get all current log segment names. Thus, we don't have to go through all index segments (total count > log segment count).